### ISSUE TITLE
Unify API to to Bcrypt.with() to be able to hash and verify from the same entrypoint

### ISSUE THREAD

**Issue by patrickfav** [patrickfav/bcrypt#26]
See discussion in #23 

> Talking about it, I think the interface of the lib a bit surprising. Why BCrypt.with()... is for hashing, and BCrypt.verifyer()... is for verifying?
It would really make more sense, from an external point of view, if we could simply do:
BCrypt.with(LongPasswordStrategies.truncate()).verify(pw, hash);

**Comment by Indigo744** [comment]
I like consistent API 😄 

Note that you can keep retro-compatibility by keeping the `BCrypt.verifyer()` as only a wrapper around `BCrypt.with()` (and marking it as obsolete, if possible in Java?).

**Comment by patrickfav** [comment]
You are right, we can make it backward compatible. And of course I agree with you, a API should be consistent :)
